core/src/main/java/org/sonarsource/sonarlint/core/websocket/WebSocketService.java

```diff
@@ -119,10 +117,21 @@ public void handleEvent(ConnectionConfigurationAddedEvent connectionConfiguratio
   public void handleEvent(ConnectionConfigurationUpdatedEvent connectionConfigurationUpdatedEvent) {
     var updatedConnectionId = connectionConfigurationUpdatedEvent.getUpdatedConnectionId();
     var connection = connectionConfigurationRepository.getConnectionById(updatedConnectionId);
-    if (connection != null && connection.getKind().equals(ConnectionKind.SONARCLOUD)) { // TODO what if connection type changed from SQ to SC or vice versa?
-      closeConnection(updatedConnectionId);
-      if (!connection.isDisableNotifications() || !connectionIdsInterestedInNotifications.isEmpty()) {
-        createConnectionIfNeeded(connection.getConnectionId());
+    if (connection != null && connection.getKind().equals(ConnectionKind.SONARCLOUD)) {
+      // TODO what if connection type changed from SQ to SC or vice versa?
+      closeSocket();
+      connectionIdsInterestedInNotifications.remove(updatedConnectionId);
+      if (!connection.isDisableNotifications()){
+        // Notifications are enabled, reopen the connection
+        createConnectionIfNeeded(updatedConnectionId);
+        resubscribeAll();
+      } else if(!connectionIdsInterestedInNotifications.isEmpty()) {
+        var configurationScopesToUnsubscribe = configurationRepository.getConfigScopesWithBindingConfiguredTo(updatedConnectionId);
+        for(var configScope : configurationScopesToUnsubscribe) {
+          subscribedProjectKeysByConfigScopes.remove(configScope.getId());
+        }
```

Comment:
I wonder if we shouldn't do this at another moment too. If there is only a single SC connection, and if it was updated to disable notifications, we wouldn't reach this branch, and the configurations copes would still be in the `subscribedProjectKeysByConfigScopes`, which might cause problems later when adding a new SC connection